inc/class-segment-transformer.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * Map-based data transformation.
+ *
+ * @package altis-analytics-integration-segment
+ */
+
+namespace Altis\Analytics\Integration\Segment\Export;
+
+/**
+ * Map-based data transformation.
+ */
+class SegmentTransformer extends Transformer {
+	/**
+	 * Maps between source and destination formats.
+	 *
+	 * @var array
+	 */
+	public $maps = [];
+
+	/**
+	 * Construct a new instance.
+	 *
+	 * @param array $map Map between source and destination formats.
+	 */
+	public function __construct() {
+		// Prepare data mapping array in advance.
+		$this->maps = [
+			'identify' => $this->get_segment_call_type_map( 'identify' ),
+			'page' => $this->get_segment_call_type_map( 'page' ),
+			'track' => $this->get_segment_call_type_map( 'track' ),
+			'groups' => apply_filters( 'altis.analytics.segment.groups', [] ),
+		];
+	}
+
+	/**
+	 * Get transformation map based on call type.
+	 *
+	 * @param string $type Segment call type.
+	 *
+	 * @return array       Transformation map.
+	 */
+	public function get_segment_call_type_map( string $type ) {
+		$mapping = [
+			'type' => $type . '|static',
+			'anonymousId' => 'endpoint.Id',
+			'messageId' => 'endpoint.RequestId',
+			'timestamp' => 'event_timestamp|' . __CLASS__ . '::format_date',
+			'userId' => 'endpoint.User.UserId',
+
+			'context' => [
+				'active' => 'endpoint.EndpointStatus',
+				'app' => [
+					'name' => '',
+					'build' => '',
+					'version' => 'endpoint.Demographics.AppVersion',
+				],
+				'campaign' => [
+					'name' => 'attributes.qv_utm_campaign',
+					'source' => 'attributes.qv_utm_source',
+					'medium' => 'attributes.qv_utm_medium',
+					'term' => 'attributes.qv_utm_term',
+					'content' => '',
```

Comment:
Can add this one in too, not sure why it's omitted here so apologies if I'm missing something.
```suggestion
					'content' => 'attributes.qv_utm_content',
```